magicctapipe/scripts/lst1_magic/lst1_magic_event_coincidence.py

```diff
@@ -143,19 +147,24 @@ def load_lst_data_file(input_file):
         inplace=True,
     )
 
+    # Read the subarray description:
+    subarray = SubarrayDescription.from_hdf(input_file)
+
     # Change the units of some parameters:
     optics = pd.read_hdf(input_file, key='configuration/instrument/telescope/optics')
     focal_length = optics['equivalent_focal_length'][0]
 
     event_data['length'] = focal_length * np.tan(np.deg2rad(event_data['length']))   # [deg] -> [m]
```

Comment:
I'm running now with something that should be compatible with this version, but I really do not understand this part of the code.
I think LST starts in units of 'm' and not 'deg', and here we apply a conversion that will effectively scale down W and L by a factor of 2. And if  there is some conversion from m to deg done earlier on and we need to convert it back, why it is only W&L and not the x, y, r, slope, kurtosis, skewness, ... ?

Reply:
Hi Julian, please find the following function that lstchain uses for computing the Hillas parameters:
https://github.com/cta-observatory/cta-lstchain/blob/7b967de9349a196c98d3832b5421b591fd25d926/lstchain/reco/r0_to_dl1.py#L133

Here the input camera geometry is defined as the CameraFrame, so the length, width and etc computed by the `hillas_parameters` function should have the unit of meter. However, there is a conversion of the unit from meter to degree only for length and width parameters (and their uncertainties) with the `_camera_distance_to_angle` function afterwards. This is why I implemented the lines to get them back to meter only for length and width, not for the other parameters (cog_x, cog_y, etc...).